addons/website_sale_slides/models/slide_channel.py

```diff
@@ -7,10 +7,16 @@
 class Channel(models.Model):
     _inherit = 'slide.channel'
 
+    def _get_default_product_id(self):
+        product_courses = self.env['product.product'].search(
+            [('detailed_type', '=', 'course')], limit=2)
+        return product_courses.id if len(product_courses) == 1 else False
+
     enroll = fields.Selection(selection_add=[
         ('payment', 'On payment')
     ], ondelete={'payment': lambda recs: recs.write({'enroll': 'invite'})})
-    product_id = fields.Many2one('product.product', 'Product', index=True)
+    product_id = fields.Many2one('product.product', 'Product', domain=[('detailed_type', '=', 'course')],
+                                 index=True, default=_get_default_product_id)
```

Comment:
While we are at it, do we really need an index ? Seems channel is not the most heavily used table ...